src/main/java/org/jsoup/select/QueryParser.java

```diff
@@ -162,6 +162,8 @@ private void findElements() {
             byId();
         else if (tq.matchChomp("."))
             byClass();
+        else if (tq.toString().endsWith("|*"))
```

Comment:
This is not the right way to do this. The TokenQueue must be consumed token by token. With this implementation, queries like `ns|* div` (or anything after the namespace wildcard selector) will fail. You will need an appropriate matcher in TokenQueue. I think you could update the current tag matcher.